R/data-gather-clean.R

```diff
@@ -0,0 +1,121 @@
+#' Get the submissions based on status
+#'
+#' Get the submissions based on status.
+#'
+#' @inheritParams mod_review_section_server
+#' @param statuses A character vector of statuses to
+#'   include from the set: "Accepted", "Rejected",
+#'   "In Review".
+#' @param group The number for a specific Synapse forms group.
+get_submissions <- function(syn, group, statuses,
+                            section_lookup_table, variable_lookup_table) {
+  if (is.null(statuses)) {
+    return(NULL)
+  }
+  submissions <- purrr::map(statuses, function(x) {
+    synapseforms::download_all_and_get_table(
+      syn = syn,
+      state_filter = x,
+      group = group
+    )
+  }) %>%
+    purrr::compact() %>% # Removes NAs
+    purrr::reduce(dplyr::full_join, by = "variables")
+
+  if (!is.null(submissions)) {
+    submissions <- make_clean_table(
+      submissions,
+      section_lookup_table,
+      variable_lookup_table
+    )
+  }
+  submissions
+}
+
+#' Make clean submissions table
+#'
+#' Make submissions table clean by fixing experiment
+#' numbering and adding in user-friendly step and label
+#' columns instead of json variable styled section and
+#' variable columns.
+#'
+#' @keywords internal
+#' @inheritParams get_submissions
+#' @inheritParams clean_experiment_variables
+make_clean_table <- function(data, section_lookup_table,
+                             variable_lookup_table) {
+  data <- synapseforms::make_tidier_table(data)
+  data <- dplyr::full_join(data, section_lookup_table, by = "section")
+  data <- clean_experiment_variables(data)
+  data <- dplyr::full_join(data, variable_lookup_table, by = "variable")
```

Comment:
As discussed in the code review meeting, let's look into making these left joins

Reply:
left_joins work great! I pushed the change. At the same time, I also moved up the removing of rows with NA responses, which greatly decreases the size of the table and should make for better performance.